### PR TITLE
Rx terms hicks branch

### PR DESCRIPTION
**Pull request by jrlegrand**
Creating pull request on behalf of @davidhicks980

### CODE REVIEW

File: dailymed-ui/src/components/RxTermsSearch/index.jsx [hunk 1]
```diff
@@ -97,46 +63,87 @@ export default class RxTermsSearch extends React.Component {
...
-                    <Button variant="primary" disabled={!selected.rxcui}>Next</Button>{' '}
-                </div>
-
-            </div>      
+          <div>
+            <Form.Group>
+              <Form.Label>Drug name</Form.Label>
+              <AsyncTypeahead
+                id="drug-name-typeahead"
+                minLength={settings.drugNameMinLength}
```
Comment (jrlegrand) [inline dailymed-ui/src/components/RxTermsSearch/index.jsx:105]: @davidhicks980 - I'm getting an error (below).  You need to add the following labelKey property to your Drug name AsyncTypeahead:  `labelKey={"displayName"}`

`Fields passed to `filterBy` should have string values. Value will be converted to a string; results may be unexpected. `

File: dailymed-ui/src/components/RxTermsSearch/index.jsx [hunk 2]
```diff
@@ -97,46 +63,87 @@ export default class RxTermsSearch extends React.Component {
...
+                .then((resp) => resp.json())
+                .then((json) => {
+                    //** Deconstruct the fetch result */
+                    const { DISPLAY_NAME: names, STRENGTHS_AND_FORMS: forms, RXCUIS: cuis } = json[2];
+                    //**If names are returned, set the dropdown options to the returned names */
+                    if (names && forms)
+                        options.drugName = names.map((name, index) => {
+                            return {
+                                displayName: name,
+                                strengthsAndForms: forms[index]
```
Comment (jrlegrand) [inline dailymed-ui/src/components/RxTermsSearch/index.jsx]: @davidhicks980 - you're missing a key piece of this function - we need to also add RXCUIs to this object so we can use them later once both drug name and drug strength/form are selected.

Comment (jrlegrand) [reply dailymed-ui/src/components/RxTermsSearch/index.jsx]: It has to be nested like this:

```
displayName
strengthsAndForms
    strengthAndForm
    rxcui
    ...
    strengthAndForm
    rxcui
...
```

Comment (jrlegrand) [reply dailymed-ui/src/components/RxTermsSearch/index.jsx:143]: I don't know WHY this works with the weird `[index][index]` but it seems to work...

```
                .then((json) => {
                    //** Deconstruct the fetch result */
                    const { DISPLAY_NAME: names, STRENGTHS_AND_FORMS: forms, RXCUIS: cuis } = json[2];
                    //**If names are returned, set the dropdown options to the returned names */
                    if (names && forms)
                        options.drugName = names.map((name, index) => {
                            return {
                                displayName: name,
                                strengthsAndForms: forms[index].map((form, index) => {
                                  return {
                                    strengthAndForm: form,
                                    rxcui: cuis[index][index]
                                  };
                                })
                            };
                        });

                    this.setState({ isLoading: false, options, selected: [] });
                });
```

Comment (jrlegrand) [reply dailymed-ui/src/components/RxTermsSearch/index.jsx]: This is kind of ugly but seems to work... thoughts?

```
                .then((json) => {
                    //** Deconstruct the fetch result */
                    const { DISPLAY_NAME: names, STRENGTHS_AND_FORMS: forms, RXCUIS: cuis } = json[2];
                    //**If names are returned, set the dropdown options to the returned names */
                    if (names && forms)
                        options.drugName = names.map((name, nameIndex) => {
                            return {
                                displayName: name,
                                strengthsAndForms: forms[nameIndex].map((form, formIndex) => {
                                  return {
                                    strengthAndForm: form,
                                    rxcui: cuis[nameIndex][formIndex]
                                  };
                                })
                            };
                        });

                    this.setState({ isLoading: false, options, selected: [] });
                });
```

**Comment by davidhicks980** [comment]
Oh man let me take a look at it in a bit. If it works it works tho 😊 David HicksPharmD Candidate | 2021UNC School of Pharmacydavid_hicks@unc.edu From: Joseph LeGrandSent: Monday, 4 January 2021 1:56 pmTo: coderxio/dailymed-uiCc: davidhicks980; MentionSubject: Re: [coderxio/dailymed-ui] Rx terms hicks branch (#3) @jrlegrand commented on this pull request.In dailymed-ui/src/components/RxTermsSearch/index.jsx:> +    */+    searchForTerms(text, settings, options) {+        if (text.length >= settings.drugNameMinLength)+            fetch(+                `https://clinicaltables.nlm.nih.gov/api/rxterms/v3/search?ef=DISPLAY_NAME,STRENGTHS_AND_FORMS,RXCUIS&maxList=&terms=${text}`+            )+                .then((resp) => resp.json())+                .then((json) => {+                    //** Deconstruct the fetch result */+                    const { DISPLAY_NAME: names, STRENGTHS_AND_FORMS: forms, RXCUIS: cuis } = json[2];+                    //**If names are returned, set the dropdown options to the returned names */+                    if (names && forms)+                        options.drugName = names.map((name, index) => {+                            return {+                                displayName: name,+                                strengthsAndForms: forms[index]I don't know WHY this works with the weird [index][index] but it seems to work...                .then((json) => {                    //** Deconstruct the fetch result */                    const { DISPLAY_NAME: names, STRENGTHS_AND_FORMS: forms, RXCUIS: cuis } = json[2];                    //**If names are returned, set the dropdown options to the returned names */                    if (names && forms)                        options.drugName = names.map((name, index) => {                            return {                                displayName: name,                                strengthsAndForms: forms[index].map((form, index) => {                                  return {                                    strengthAndForm: form,                                    rxcui: cuis[index][index]                                  };                                })                            };                        });                     this.setState({ isLoading: false, options, selected: [] });                });—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by davidhicks980** [comment]
That makes more sense then. I was wondering where the rxcuis came in because I was working on the async typeahead exclusively? Maybe... it was a little while ago. I think the dual index is fine tbh...performance is so fast that it shouldn’t matter even if it isn’t the fastest way to do things (it may be but I can’t think that much). So in this case, should I pull from your branch or my branch or how does that work?  David HicksPharmD Candidate | 2021UNC School of Pharmacydavid_hicks@unc.edu From: Joseph LeGrandSent: Monday, 4 January 2021 1:58 pmTo: coderxio/dailymed-uiCc: davidhicks980; MentionSubject: Re: [coderxio/dailymed-ui] Rx terms hicks branch (#3) @jrlegrand commented on this pull request.In dailymed-ui/src/components/RxTermsSearch/index.jsx:> +    */+    searchForTerms(text, settings, options) {+        if (text.length >= settings.drugNameMinLength)+            fetch(+                `https://clinicaltables.nlm.nih.gov/api/rxterms/v3/search?ef=DISPLAY_NAME,STRENGTHS_AND_FORMS,RXCUIS&maxList=&terms=${text}`+            )+                .then((resp) => resp.json())+                .then((json) => {+                    //** Deconstruct the fetch result */+                    const { DISPLAY_NAME: names, STRENGTHS_AND_FORMS: forms, RXCUIS: cuis } = json[2];+                    //**If names are returned, set the dropdown options to the returned names */+                    if (names && forms)+                        options.drugName = names.map((name, index) => {+                            return {+                                displayName: name,+                                strengthsAndForms: forms[index]This is kind of ugly but seems to work... thoughts?                .then((json) => {                    //** Deconstruct the fetch result */                    const { DISPLAY_NAME: names, STRENGTHS_AND_FORMS: forms, RXCUIS: cuis } = json[2];                    //**If names are returned, set the dropdown options to the returned names */                    if (names && forms)                        options.drugName = names.map((name, nameIndex) => {                            return {                                displayName: name,                                strengthsAndForms: forms[nameIndex].map((form, formIndex) => {                                  return {                                    strengthAndForm: form,                                    rxcui: cuis[nameIndex][formIndex]                                  };                                })                            };                        });                     this.setState({ isLoading: false, options, selected: [] });                });—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by jrlegrand** [comment]
@davidhicks980 - I pushed my changes to the branch since I already did the work locally - just to save time.

**Review by davidhicks980 — APPROVED** [review]
Looks good. Good trick with the dual map. I looked up alternative ways to do it but they end up being just as complex (they use .reduce), so no worries there.

**Pull request merged**